dashboard-widgets.el

```diff
@@ -269,6 +271,88 @@ If MESSAGEBUF is not nil then MSG is also written in message buffer."
 	  (dashboard-insert--shortcut "p" "Projects:")))
     (error "Projects list depends on 'projectile-mode` to be activated")))
 
+;;
+;; Org Agenda
+;;
+(defun dashboard-insert-agenda-list (list-display-name list)
+  "Render LIST-DISPLAY-NAME title and agenda items from LIST."
+  (when (car list)
+    (insert list-display-name)
+    (mapc (lambda (el)
+            (insert "\n    ")
+            (let ((filename (nth 4 el))
+                  (lineno (nth 3 el))
+                  (title (nth 0 el)))
+              (widget-create 'push-button
+                             :action `(lambda (&rest ignore)
+                                        (let ((buffer (find-file-other-window ,filename)))
+                                          (with-current-buffer buffer
+                                            (goto-char ,lineno)
+                                            )
+                                          (switch-to-buffer buffer)))
+                             :mouse-face 'highlight
+                             :follow-link "\C-m"
+                             :button-prefix ""
+                             :button-suffix ""
+                             :format "%[%t%]"
+                             (format "%s" title)))
+            )
+          list)))
+
+(defun dashboard--timestamp-to-gregorian-date (timestamp)
+  "Convert TIMESTAMP to a gregorian date.
+
+The result can be used with functions like
+`calendar-date-compare'."
+  (let ((decoded-timestamp (decode-time timestamp)))
+    (list (nth 4 decoded-timestamp)
+          (nth 3 decoded-timestamp)
+          (nth 5 decoded-timestamp))))
+
+(defun dashboard--date-due-p (timestamp &optional due-date)
+  "Check if TIMESTAMP is today or in the past.
+
+If DUE-DATE is nil, compare TIMESTAMP to today; otherwise,
+compare to the date in DUE-DATE.
+
+The time part of both TIMESTAMP and DUE-DATE is ignored, only the
+date part is considered."
+  (unless due-date
+    (setq due-date (current-time)))
+  (let* ((gregorian-date (dashboard--timestamp-to-gregorian-date timestamp))
+         (gregorian-due-date (dashboard--timestamp-to-gregorian-date due-date)))
+    (calendar-date-compare (list gregorian-date)
+                           (list gregorian-due-date))))
+
+(defun dashboard--get-agenda ()
+  "Get agenda items for today."
+  (require 'org-agenda)
```

Comment:
I went the same way as `dashboard-insert-bookmarks` does. Again, thanks for fixing it!